en, zh: add release notes for v1.2.0 (#1328)
pingcap/docs-tidb-operator: en/releases/release-1.2.0.md

---
title: TiDB Operator 1.2.0 Release Notes
---

# TiDB Operator 1.2.0 Release Notes

Release date: July 29, 2021

TiDB Operator version: 1.2.0

## Rolling update changes

- Upgrading TiDB Operator will cause the recreation of the TidbMonitor Pod due to [#4085](https://github.com/pingcap/tidb-operator/pull/4085)

## New features

> - Support setting Prometheus `retentionTime`, which is more fine-grained than `reserveDays`, and only use `retentionTime` if both are configured ([#4085](https://github.com/pingcap/tidb-operator/pull/4085), [@better0332](https://github.com/better0332))

Reviewer: ```suggestion
- Support setting Prometheus `retentionTime`, which is more fine-grained than `reserveDays`, and only `retentionTime` takes effect if both are configured ([#4085](https://github.com/pingcap/tidb-operator/pull/4085), [@better0332](https://github.com/better0332))
```

> - Support setting `retentionTime` for Prometheus with more fine-grained duration than `reserveDays`, and will only use `retentionTime` if both configured ([#4085](https://github.com/pingcap/tidb-operator/pull/4085), [@better0332](https://github.com/better0332))

Reviewer: ```suggestion
- Support setting Prometheus `retentionTime`, which is more fine-grained than `reserveDays`, and only use `retentionTime` if both are configured ([#4085](https://github.com/pingcap/tidb-operator/pull/4085), [@better0332](https://github.com/better0332))
```

> - Support setting `priorityClassName` of `Backup` Job ([#4078](https://github.com/pingcap/tidb-operator/pull/4078), [@mikechengwei](https://github.com/mikechengwei))

Reviewer 2: ```suggestion
- Support setting `priorityClassName` in the `Backup` CR to specify the priority of the backup Job ([#4078](https://github.com/pingcap/tidb-operator/pull/4078), [@mikechengwei](https://github.com/mikechengwei))
```

> - Support setting `priorityClassName` in the `Backup` CR to specify the priority of the backup Job ([#4078](https://github.com/pingcap/tidb-operator/pull/4078), [@mikechengwei](https://github.com/mikechengwei))

Reviewer: ```suggestion
- Support setting `priorityClassName` in the `Backup` CR to specify the priority of the backup job ([#4078](https://github.com/pingcap/tidb-operator/pull/4078), [@mikechengwei](https://github.com/mikechengwei))
```

## Improvements

> - Set the default region leader eviction timeout of TiKV to 1500 min ([#4071](https://github.com/pingcap/tidb-operator/pull/4071), [@KanShiori](https://github.com/KanShiori))

Reviewer: ```suggestion
- Changes the default region leader eviction timeout of TiKV to 1500 minutes for TiKV graceful upgrade ([#4071](https://github.com/pingcap/tidb-operator/pull/4071), [@KanShiori](https://github.com/KanShiori))
```

> - Changes the default region leader eviction timeout of TiKV to 1500 minutes to prevent the Pod from being deleted when the region leaders are not transferred completely to the other stores which will cause data corruption([#4071](https://github.com/pingcap/tidb-operator/pull/4071), [@KanShiori](https://github.com/KanShiori))

Reviewer: ```suggestion
- Changes the default Region leader eviction timeout of TiKV to 1500 minutes. The change prevents the Pod from being deleted when the Region leaders are not transferred completely to the other stores, which will cause data corruption ([#4071](https://github.com/pingcap/tidb-operator/pull/4071), [@KanShiori](https://github.com/KanShiori))
```

## Bug fixes

> - Fix the issue that may parse URL of `Prometheus.RemoteWrite` in `TiDBMonitor` failed ([#4087](https://github.com/pingcap/tidb-operator/pull/4087), [@better0332](https://github.com/better0332))

Reviewer 2: ```suggestion
- Fix the issue that it may fail to parse the URL in `Prometheus.RemoteWrite` in `TiDBMonitor` ([#4087](https://github.com/pingcap/tidb-operator/pull/4087), [@better0332](https://github.com/better0332))
```

> - Fix the issue that it may fail to parse the URL in `Prometheus.RemoteWrite` in `TiDBMonitor` ([#4087](https://github.com/pingcap/tidb-operator/pull/4087), [@better0332](https://github.com/better0332))

Reviewer: it 这里不太清楚，请问是 TiDB Operator 吗？

Author: 这里 it 类似于代码？修复一个代码里的问题

Author: 应该就算 Operator

Reviewer 2: ```suggestion
- Fix the issue that the URL in `Prometheus.RemoteWrite` may be parsed incorrectly in `TiDBMonitor` ([#4087](https://github.com/pingcap/tidb-operator/pull/4087), [@better0332](https://github.com/better0332))
```
